src/ObjWriting/Game/IW5/LeaderboardDef/JsonLeaderboardDefWriter.cpp

```diff
@@ -0,0 +1,97 @@
+#include "JsonLeaderboardDefWriter.h"
+
+#include "Game/IW5/CommonIW5.h"
+#include "Game/IW5/Json/JsonLeaderboardDef.h"
+
+#include <iomanip>
+#include <nlohmann/json.hpp>
+
+using namespace nlohmann;
+using namespace IW5;
+
+namespace
+{
+    class JsonDumper
+    {
+    public:
+        explicit JsonDumper(std::ostream& stream)
+            : m_stream(stream)
+        {
+        }
+
+        void Dump(const LeaderboardDef* leaderboardDef)
+        {
+            JsonLeaderboardDef jsonLeaderboardDef;
+            CreateJsonLeaderboardDef(jsonLeaderboardDef, *leaderboardDef);
+
+            json jRoot = jsonLeaderboardDef;
+
+            jRoot["_type"] = "leaderboardDef";
+            jRoot["_version"] = 1;
+
+            m_stream << std::setw(4) << jRoot << "\n";
+        }
+
+    private:
+        static const char* AssetName(const char* input)
+        {
+            if (input && input[0] == ',')
+                return &input[1];
+
+            return input;
+        }
+
+        static void CreateJsonColumnDef(JsonColumnDef& jColumnDef, const LbColumnDef& lbColumnDef)
+        {
+            if (lbColumnDef.name)
+            {
+                jColumnDef.name = AssetName(lbColumnDef.name);
+            }
+
+            jColumnDef.id = lbColumnDef.id;
+            jColumnDef.propertyId = lbColumnDef.propertyId;
+            jColumnDef.hidden = lbColumnDef.hidden;
+
+            if (lbColumnDef.statName)
+            {
+                jColumnDef.statName = AssetName(lbColumnDef.statName);
```

Comment:
`statName` is not an asset name 😅 This function is just meant to resolve asset references. Same goes for `name` but that should not be a property anyway, see above.

In fact the AssetName function should not be used anywhere here, since the Leaderboard does not reference any other assets.